python/cli.rst

```diff
@@ -0,0 +1,68 @@
+##############################
+Python Command Line Interfaces
+##############################
+
+Command line interfaces (CLI) in Python may be implemented using the ``argparse`` module or using `Click`_.
+
+.. _Click: https://click.palletsprojects.com
+
+.. _cli-using-click:
+
+Command Line Interfaces using Click
+===================================
+
+Click is a Python package for creating command line interfaces in a composable way. It supports:
+
+* arbitrary nesting of commands
+* automatic help page generation
+* lazy loading of subcommands at runtime
+
+`Click`_ has good `documentation`_ including a `quickstart`_ guide.
+
+.. _documentation: https://click.palletsprojects.com/en/7.x/#documentation
+.. _quickstart: https://click.palletsprojects.com/en/7.x/quickstart/
+
+The ``butler`` CLI command is implemented in ``daf_butler``. It has a reusalbe framework for dynamically
+loading subcommands from plugin packages. It is documented in the ``daf_butler`` package documentation.
+
+Add a Callable CLI Command To Your Package
+==========================================
+
+To create a callable command at the top level of your package create a folder called ``bin.src``, it should
+contain two files:
+
+1. ``SConscript`` with contents:
```

Comment:
The indentation in this list is a little off, which makes the rendering look wrong — in the attached screenshot, compare the version on the left with my re-indented version on the right. 
![Screenshot 2020-08-27 at 10 32 03](https://user-images.githubusercontent.com/460854/91475535-969b2880-e850-11ea-9477-a1e1361b4da9.png)

The diff is:
```
diff --git a/python/cli.rst b/python/cli.rst
index 5f9939f..4dcfe0b 100644
--- a/python/cli.rst
+++ b/python/cli.rst
@@ -33,36 +33,33 @@ contain two files:

 1. ``SConscript`` with contents:

-.. code-block:: py
+   .. code-block:: py

-    # -*- python -*-
-    from lsst.sconsUtils import scripts
-    scripts.BasicSConscript.shebang()
+       # -*- python -*-
+       from lsst.sconsUtils import scripts
+       scripts.BasicSConscript.shebang()

 2. A file that has the name of the CLI command the user will call. This file should contain as little
    implementation as possible, it should import an implementation file and call a function in it. For example:

-    a. **Click Scripts**
+   a. **Click Scripts**

-    A file ``bin.src/butler`` is implemented as:
+      A file ``bin.src/butler`` is implemented as:

-    .. code-block:: py
+      .. code-block:: py

-        import sys
-        from lsst.daf.butler.cli.butler import main
+          import sys
+          from lsst.daf.butler.cli.butler import main

-        if __name__ == '__main__':
-            sys.exit(main())
+          if __name__ == '__main__':
+              sys.exit(main())

-    b. **Pipe Task Scripts**
+   b. **Pipe Task Scripts**

-    A file ``bin.src/calibrate.py`` is implemented as:
-
-    .. code-block:: py
-
-        from lsst.pipe.tasks.calibrate import CalibrateTask
-
-        CalibrateTask.parseAndRun()
+      A file ``bin.src/calibrate.py`` is implemented as:

+      .. code-block:: py

+          from lsst.pipe.tasks.calibrate import CalibrateTask

+          CalibrateTask.parseAndRun()
```